### PR TITLE
Allow customizing the Application Context builder

### LINKED ISSUE #392: Unable to customize ApplicationContext on startup
### Expected Behavior

When specifying an `ApplicationContextConfigurer`, I should be able to modify the Application Context during construction

```
import io.micronaut.context.ApplicationContextBuilder
import io.micronaut.context.ApplicationContextConfigurer
import io.micronaut.context.annotation.ContextConfigurer

@ContextConfigurer
class Configurer: ApplicationContextConfigurer {
    override fun configure(builder: ApplicationContextBuilder) {
        builder.eagerInitSingletons(true)
    }
}
```

Or per the documentation, since we extend the `AzureFunction` class, there should be some way to customize the application context via an override - https://docs.micronaut.io/latest/guide/#eagerInit

### Actual Behaviour

There is no way to configure eager initialization of Singletons

### Steps To Reproduce

_No response_

### Environment Information

_No response_

### Example Application

_No response_

### Version

3.7.3

### PR DESCRIPTION
**Pull request by BenD10**
Fixes #392

Allows us to customize the application context, specifically in our case so we can eagerly initialise singletons on our premium Azure function. Aligns with the existing documentation in https://docs.micronaut.io/latest/guide/#eagerInit

### CODE REVIEW

**Comment by graemerocher** [comment]
PR doesn't compile

**Comment by BenD10** [comment]
My apologises @graemerocher, it should be fixed now.

File: azure-function/src/main/java/io/micronaut/azure/function/AzureFunction.java [hunk 1]
```diff
@@ -65,10 +66,12 @@ protected AzureFunction() {
         applicationContext.inject(this);
     }
 
+    protected static ApplicationContextBuilder newApplicationContextBuilder() {
```
Comment (graemerocher) [inline azure-function/src/main/java/io/micronaut/azure/function/AzureFunction.java:69]: method needs javadoc and please annotate the return type with `@NonNull`

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Comment by graemerocher** [comment]
Thanks!

**Pull request merged**